### PR TITLE
Do not suggest nonsensical OpenSSL verify modes

### PR DESCRIPTION
**Pull request by jhass**
`SSL_set_verify(3)` explains:

```
SSL_VERIFY_FAIL_IF_NO_PEER_CERT
  Server mode: if the client did not return a certificate, the TLS/SSL
handshake is immediately terminated with a "handshake failure" alert.
This flag must
  be used together with SSL_VERIFY_PEER.

  Client mode: ignored

SSL_VERIFY_CLIENT_ONCE
  Server mode: only request a client certificate on the initial TLS/SSL
handshake. Do not ask for a client certificate again in case of a
renegotiation.
  This flag must be used together with SSL_VERIFY_PEER.

  Client mode: ignored
```

The SMTP connection here uses a OpenSSL socket in client mode,
suggesting invalid/ignored flags is rather misleading.

### CODE REVIEW

File: guides/source/configuring.md [hunk 1]
```diff
@@ -524,7 +524,7 @@ There are a number of settings available on `config.action_mailer`:
     * `:password` - If your mail server requires authentication, set the password in this setting.
     * `:authentication` - If your mail server requires authentication, you need to specify the authentication type here. This is a symbol and one of `:plain`, `:login`, `:cram_md5`.
     * `:enable_starttls_auto` - Detects if STARTTLS is enabled in your SMTP server and starts to use it. It defaults to `true`.
-    * `:openssl_verify_mode` - When using TLS, you can set how OpenSSL checks the certificate. This is useful if you need to validate a self-signed and/or a wildcard certificate. This can be one of the OpenSSL verify constants, `:none`, `:peer`, `:client_once`, `:fail_if_no_peer_cert`, or the constant directly `OpenSSL::SSL::VERIFY_NONE`.
+    * `:openssl_verify_mode` - When using TLS, you can set how OpenSSL checks the certificate. This is useful if you need to validate a self-signed and/or a wildcard certificate. This can be one of the OpenSSL verify constants, `:none` or `:peer`, or the constant directly `OpenSSL::SSL::VERIFY_NONE` or `OpenSSL::SSL::VERIFY_PEER`.
```
Comment (maclover7) [inline guides/source/configuring.md]: [...] You can use either the symbols, `:none` and `:peer`, or the constants directly -- `OpenSSL::SSL::VERIFY_NONE` and `OpenSSL::SSL::VERIFY_PEER`, respectively.

**Comment by maclover7** [comment]
Does the `OpenSSL` stuff appear anywhere else in Rails' docs?

**Comment by jhass** [comment]
I grepped for `client_once`, `fail_if_no_peer_cert` and `OpenSSL::SSL::VERIFY` and couldn't spot any further occurrences.

**Comment by maclover7** [comment]
👍 cool.

Can you squash your commits, and add `[ci skip]` to avoid running the test suite? Thanks!

**Comment by jhass** [comment]
Sure, done.

**Comment by vipulnsward** [comment]
Thanks @jhass , would have been good to point to ruby-docs for this, but none exist. 
Will be a good ruby doc change, as well.

**Pull request merged**